go/v1beta1/storage/filtering/filtering.go

```diff
@@ -46,8 +47,14 @@ func parseExpression(expression *expr.Expr) (*Query, error) {
 	function := expression.GetCallExpr().GetFunction()
 
 	// Determine if left and right side are final and if so formulate query
-	leftArg := expression.GetCallExpr().Args[0]
-	rightArg := expression.GetCallExpr().Args[1]
+	var leftArg, rightArg *expr.Expr
+	if len(expression.GetCallExpr().Args) == 2 {
```

Comment:
For an expression `a == b`, `a` and `b` are treated as arguments to the `_==_` operator; but in an expression `a.startsWith(b)`, `a` is the target/receiver and `b` is the argument.

Reply:
can you add this explanation as a comment here?  we can remove it later once we figure out a better way for this